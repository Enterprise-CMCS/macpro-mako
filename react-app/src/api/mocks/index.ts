export * as mockSubmit from "./submit";
export * as mockTypes from "./types";
