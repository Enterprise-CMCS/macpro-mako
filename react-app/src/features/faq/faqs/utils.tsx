import { cn } from "@/utils";

export interface Template {
  title: string;
  text?: string;
  href: string;
  subtext?: string[];
}

export const helpDeskContact = {
  email: "OneMAC_Helpdesk@cms.hhs.gov",
  phone: "(833) 228-2540",
};

export const slugify = (text: string) =>
  text
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, "-")
    .replace(/(^-|-$)/g, "")
    .substring(0, 40);

export const handleSupportLinkClick = (type: string) => (e: React.MouseEvent<HTMLElement>) => {
  const text = e.currentTarget.textContent?.trim() || "unknown";
  window.gtag("event", `support_click_${type}_${slugify(text)}`, {
    event_category: "Support",
    event_label: text,
  });
};

export const PdfLink = ({
  href,
  title,
  text,
  label,
  className = "underline hover:no-underline",
}: {
  href: string;
  title: string;
  text?: string;
  label: string;
  className?: string;
}) => (
  <a
    className={cn("text-primary", className)}
    href={href}
    download
    rel="noopener noreferrer"
    target="_blank"
    onClick={handleSupportLinkClick(label)}
  >
    {title}
    {text && `: ${text}`}
  </a>
);

export const PdfList = ({
  list,
  label,
  ulClassName = "list-disc pl-6 space-y-2",
}: {
  list: Template[];
  label: string;
  ulClassName?: string;
}) => (
  <ul className={ulClassName} role="list">
    {list.map((pdf) => (
      <li key={pdf.title}>
        <PdfLink href={pdf.href} title={pdf.title} text={pdf.text} label={label} className="" />
        {pdf.subtext && (
          <ul className="list-disc pl-7 space-y-1" role="list">
            {pdf.subtext.map((sub, index) => (
              <li key={index} className="text-sm text-gray-600">
                {sub}
              </li>
            ))}
          </ul>
        )}
      </li>
    ))}
  </ul>
);
