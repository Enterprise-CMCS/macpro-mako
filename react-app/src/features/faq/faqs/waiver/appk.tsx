export const Appk = () => (
  <p>Yes, you can submit Appendix K amendments in the 1915(c) Appendix K form.</p>
);
