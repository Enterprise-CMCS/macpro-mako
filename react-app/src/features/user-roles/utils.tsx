export type StatusType = "active" | "pending" | "denied" | "revoked";
export type UserRoleType = {
  id: string;
  email: string;
  fullName: string;
  role: string;
  territory: string;
  doneByEmail: string;
  doneByName: string;
  lastModifiedDate: number;
  status: StatusType;
  eventType: string;
  group?: string;
  division?: string;
};

export const initSortUserData = (userData: UserRoleType[]) => {
  if (!userData.length) return [];
  // separate pending / other
  const pendingRoles = userData.filter((x: UserRoleType) => x.status === "pending");
  const remainingRoles = userData.filter((x: UserRoleType) => x.status !== "pending");

  const compare = (a: UserRoleType, b: UserRoleType) => {
    const nameA = a.fullName.toLocaleLowerCase();
    const nameB = b.fullName.toLocaleLowerCase();

    if (nameA < nameB) return -1;
    if (nameA > nameB) return 1;
    return 0;
  };
  const sorted = pendingRoles.sort(compare).concat(remainingRoles.sort(compare));

  return sorted;
};

export const sortUserData = (
  sortByKey: keyof UserRoleType,
  direction: boolean,
  data: UserRoleType[],
) => {
  if (!data.length) return [];

  // when direction is true, that means we are descending
  const [last, first] = direction ? [-1, 1] : [1, -1];
  const sortStatus = (a: UserRoleType, b: UserRoleType) => {
    switch (a.status) {
      case "pending":
        return first;
      case "active":
        return b.status === "pending" ? last : first;
      case "denied":
        return b.status === "revoked" ? first : last;
      case "revoked":
        return last;
      default:
        return last;
    }
  };
  const mainSort = (a: UserRoleType, b: UserRoleType) =>
    a[sortByKey] < b[sortByKey] ? first : last;

  if (sortByKey === "status") return data.sort((a, b) => sortStatus(a, b));

  return data.sort((a, b) => mainSort(a, b));
};
