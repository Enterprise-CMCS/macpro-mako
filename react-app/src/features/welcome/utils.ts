import { QueryClient } from "@tanstack/react-query";
import { redirect } from "react-router";

import { getUser } from "@/api";

export const loader = (queryClient: QueryClient) => {
  return async () => {
    const queryString = window.location.search;
    // Parse the query string to get URL parameters
    const queryParams = new URLSearchParams(queryString);
    // Access specific parameters
    const errorDescription = queryParams.get("error_description");
    const error = queryParams.get("error");
    if (errorDescription || error) {
      console.error("Authentication Error:", { errorDescription, error });
      return { error };
    }

    // check user query has been initialized
    if (!queryClient.getQueryData(["user"])) {
      const userFetch = await queryClient.fetchQuery({
        queryKey: ["user"],
        queryFn: () => getUser(),
      });
      const isUserLoggedIn =
        !["/login", "/faq"].includes(window.location.pathname) && !userFetch?.user;

      if (isUserLoggedIn) {
        return redirect("/login");
      }
    }

    return queryClient.getQueryData(["user"]);
  };
};
