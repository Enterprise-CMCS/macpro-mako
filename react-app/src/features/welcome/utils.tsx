import { QueryClient } from "@tanstack/react-query";
import { Navigate, redirect } from "react-router";

import { getUser, OneMacUser } from "@/api";

export const loader = (queryClient: QueryClient) => {
  return async () => {
    const queryString = window.location.search;
    // Parse the query string to get URL parameters
    const queryParams = new URLSearchParams(queryString);
    // Access specific parameters
    const errorDescription = queryParams.get("error_description");
    const error = queryParams.get("error");
    if (errorDescription || error) {
      console.error("Authentication Error:", { errorDescription, error });
      return { error };
    }

    // check user query has been initialized
    if (!queryClient.getQueryData(["user"])) {
      const userFetch = await queryClient.fetchQuery({
        queryKey: ["user"],
        queryFn: () => getUser(),
      });
      return !userFetch?.user && redirect("/login");
    }

    // check user is logged in
    const loginRedirect =
      !["/login", "/faq", "/support"].includes(window.location.pathname) &&
      !queryClient.getQueryData<OneMacUser>(["user"])?.user;
    if (loginRedirect) {
      return <Navigate to="/login" />;
    }

    return queryClient.getQueryData(["user"]);
  };
};
