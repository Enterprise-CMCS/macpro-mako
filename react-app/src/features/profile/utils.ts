import { UserRole } from "shared-types/events/legacy-user";

import { StateAccess } from "@/api";
import { convertStateAbbrToFullName } from "@/utils";

export const userRoleMap = {
  defaultcmsuser: "CMS Read-only User",
  cmsroleapprover: "CMS Role Approver",
  cmsreviewer: "CMS Read-only User",
  statesystemadmin: "State System Admin",
  helpdesk: "Helpdesk",
  statesubmitter: "State Submitter",
  systemadmin: "System Admin",
};

export const stateAccessRoles: UserRole[] = [
  "statesubmitter",
  "statesystemadmin",
  "cmsroleapprover",
  "systemadmin",
];

export const orderStateAccess = (accesses: StateAccess[]) => {
  if (!accesses || !accesses.length) return;
  // sort revoked states seprately and add to
  const activeStates = accesses.filter((x: StateAccess) => x.status != "revoked");
  const revokedStates = accesses.filter((x: StateAccess) => x.status == "revoked");

  const compare = (a: StateAccess, b: StateAccess) => {
    const stateA = convertStateAbbrToFullName(a.territory);
    const stateB = convertStateAbbrToFullName(b.territory);

    if (stateA < stateB) return -1;
    if (stateA > stateB) return 1;
    return 0;
  };

  const sorted = activeStates.sort(compare).concat(revokedStates.sort(compare));

  return sorted;
};

// if user has no active roles, show pending state(s)
// show state(s) for latest active role
export const filterStateAccess = (userDetails, userProfile) => {
  if (!userProfile?.stateAccess || userProfile.stateAccess.length < 1) return [];
  return userDetails?.role
    ? userProfile.stateAccess.filter(
        (access: StateAccess) => access.role === userDetails.role && access.territory !== "ZZ",
      )
    : userProfile.stateAccess.filter((access: StateAccess) => access.territory !== "ZZ");
};
