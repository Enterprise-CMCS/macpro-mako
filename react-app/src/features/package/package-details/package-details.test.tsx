import { describe, expect, it, vi } from "vitest";
import { UseQueryResult } from "@tanstack/react-query";

import * as api from "@/api/useGetUser";
import * as gi from "@/api/useGetItem";

import { PackageDetails } from ".";
import { AppK } from "./appk";

import { mockUseGetUser, WITHDRAW_APPK_ITEM, WITHDRAW_APPK_ITEM_ID } from "mocks";
import { renderWithQueryClient } from "@/utils/test-helpers";
import { getItem, OneMacUser } from "@/api";

describe("package details", () => {
  vi.spyOn(gi, "useGetItemCache").mockReturnValue({
    data: WITHDRAW_APPK_ITEM._source,
    refetch: vi.fn(),
  });

  it("makes a package", async () => {
    vi.spyOn(api, "useGetUser").mockImplementation(() => {
      const response = mockUseGetUser();
      response.data.isCms = false;
      return response as UseQueryResult<OneMacUser, unknown>;
    });
    const renderComponent = async () => {
      const item = await getItem(WITHDRAW_APPK_ITEM_ID);
      return renderWithQueryClient(<PackageDetails itemResult={item} />);
    };
    const compo = await renderComponent();
    expect(compo).toMatchSnapshot();
  });
  it("makes a an appk element", async () => {
    vi.spyOn(api, "useGetUser").mockImplementation(() => {
      const response = mockUseGetUser();
      response.data.isCms = false;
      return response as UseQueryResult<OneMacUser, unknown>;
    });
    const renderComponent = async () => {
      return renderWithQueryClient(<AppK />);
    };
    const compo = await renderComponent();
    expect(compo).toMatchSnapshot();
  });
});
