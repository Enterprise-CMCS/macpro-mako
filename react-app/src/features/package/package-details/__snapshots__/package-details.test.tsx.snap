// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`package details > makes a an appk element 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-2"
        id="appendix_k"
      >
        <div
          class="w-full border-[1px] overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <thead
              class="[&_tr]:border-b"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
              >
                <th
                  class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5 w-[300px]"
                >
                  <div
                    class="flex items-center gap-1"
                  >
                    Package ID
                    <svg
                      class=".1em w-5 h-5 opacity-0"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m5 12 7-7 7 7"
                      />
                      <path
                        d="M12 19V5"
                      />
                    </svg>
                  </div>
                </th>
                <th
                  class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5"
                >
                  <div
                    class="flex items-center gap-1"
                  >
                    Actions
                    <svg
                      class=".1em w-5 h-5 opacity-0"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m5 12 7-7 7 7"
                      />
                      <path
                        d="M12 19V5"
                      />
                    </svg>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
              >
                <td
                  class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0 font-medium"
                >
                  <p>
                    withdrawn
                  </p>
                </td>
                <td
                  class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0"
                >
                  <button
                    class="items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-9 rounded-md px-3 flex gap-1"
                  >
                    <svg
                      class="lucide lucide-undo2"
                      fill="none"
                      height="20"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 14 4 9l5-5"
                      />
                      <path
                        d="M4 9h10.5a5.5 5.5 0 0 1 5.5 5.5v0a5.5 5.5 0 0 1-5.5 5.5H11"
                      />
                    </svg>
                    Withdraw
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="my-2"
      id="appendix_k"
    >
      <div
        class="w-full border-[1px] overflow-auto"
      >
        <table
          class="w-full caption-bottom text-sm"
        >
          <thead
            class="[&_tr]:border-b"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
            >
              <th
                class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5 w-[300px]"
              >
                <div
                  class="flex items-center gap-1"
                >
                  Package ID
                  <svg
                    class=".1em w-5 h-5 opacity-0"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m5 12 7-7 7 7"
                    />
                    <path
                      d="M12 19V5"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5"
              >
                <div
                  class="flex items-center gap-1"
                >
                  Actions
                  <svg
                    class=".1em w-5 h-5 opacity-0"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m5 12 7-7 7 7"
                    />
                    <path
                      d="M12 19V5"
                    />
                  </svg>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
            >
              <td
                class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0 font-medium"
              >
                <p>
                  withdrawn
                </p>
              </td>
              <td
                class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0"
              >
                <button
                  class="items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-9 rounded-md px-3 flex gap-1"
                >
                  <svg
                    class="lucide lucide-undo2"
                    fill="none"
                    height="20"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 14 4 9l5-5"
                    />
                    <path
                      d="M4 9h10.5a5.5 5.5 0 0 1 5.5 5.5v0a5.5 5.5 0 0 1-5.5 5.5H11"
                    />
                  </svg>
                  Withdraw
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`package details > makes a package 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="p-4 border rounded-sm border-cardBorder"
        id="package_details"
      >
        <h1
          class="text-3xl font-semibold mb-2"
          data-testid="undefined-title"
        >
          1915(c) Package Details
        </h1>
        <hr
          class="my-6 bg-slate-200"
        />
        <div
          class="gap-8 flex flex-col"
          data-testid="undefined-child"
        >
          <div
            class="flex-col gap-4 max-w-2xl"
          >
            <div
              class="grid grid-cols-2 gap-6 py-4"
            >
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Submission ID
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  MD-010.R00.02
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Authority
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  1915(c)
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Action Type
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  Initial
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  State
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  Maryland
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Proposed effective date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  Pending
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Initial submission date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Latest package activity
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Formal RAI response date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Final disposition date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Approved effective date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
            </div>
            <hr
              class="my-4"
            />
            <div
              class="grid grid-cols-2 gap-6"
            >
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Submitted by
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  <p
                    class="text-lg"
                  >
                    -- --
                  </p>
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  CPOC
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  <p
                    class="text-lg"
                  >
                    -- --
                  </p>
                </div>
              </div>
            </div>
            <div
              class="my-2"
              id="appendix_k"
            >
              <div
                class="w-full border-[1px] overflow-auto"
              >
                <table
                  class="w-full caption-bottom text-sm"
                >
                  <thead
                    class="[&_tr]:border-b"
                  >
                    <tr
                      class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
                    >
                      <th
                        class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5 w-[300px]"
                      >
                        <div
                          class="flex items-center gap-1"
                        >
                          Package ID
                          <svg
                            class=".1em w-5 h-5 opacity-0"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m5 12 7-7 7 7"
                            />
                            <path
                              d="M12 19V5"
                            />
                          </svg>
                        </div>
                      </th>
                      <th
                        class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5"
                      >
                        <div
                          class="flex items-center gap-1"
                        >
                          Actions
                          <svg
                            class=".1em w-5 h-5 opacity-0"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m5 12 7-7 7 7"
                            />
                            <path
                              d="M12 19V5"
                            />
                          </svg>
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="[&_tr:last-child]:border-0"
                  >
                    <tr
                      class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
                    >
                      <td
                        class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0 font-medium"
                      >
                        <p>
                          withdrawn
                        </p>
                      </td>
                      <td
                        class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0"
                      >
                        <button
                          class="items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-9 rounded-md px-3 flex gap-1"
                        >
                          <svg
                            class="lucide lucide-undo2"
                            fill="none"
                            height="20"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M9 14 4 9l5-5"
                            />
                            <path
                              d="M4 9h10.5a5.5 5.5 0 0 1 5.5 5.5v0a5.5 5.5 0 0 1-5.5 5.5H11"
                            />
                          </svg>
                          Withdraw
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </section>
      ,
    </div>
  </body>,
  "container": <div>
    <section
      class="p-4 border rounded-sm border-cardBorder"
      id="package_details"
    >
      <h1
        class="text-3xl font-semibold mb-2"
        data-testid="undefined-title"
      >
        1915(c) Package Details
      </h1>
      <hr
        class="my-6 bg-slate-200"
      />
      <div
        class="gap-8 flex flex-col"
        data-testid="undefined-child"
      >
        <div
          class="flex-col gap-4 max-w-2xl"
        >
          <div
            class="grid grid-cols-2 gap-6 py-4"
          >
            <div>
              <h3
                style="font-weight: 700;"
              >
                Submission ID
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                MD-010.R00.02
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Authority
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                1915(c)
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Action Type
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                Initial
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                State
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                Maryland
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Proposed effective date
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                Pending
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Initial submission date
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                -- --
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Latest package activity
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                -- --
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Formal RAI response date
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                -- --
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Final disposition date
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                -- --
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                Approved effective date
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                -- --
              </div>
            </div>
          </div>
          <hr
            class="my-4"
          />
          <div
            class="grid grid-cols-2 gap-6"
          >
            <div>
              <h3
                style="font-weight: 700;"
              >
                Submitted by
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                <p
                  class="text-lg"
                >
                  -- --
                </p>
              </div>
            </div>
            <div>
              <h3
                style="font-weight: 700;"
              >
                CPOC
              </h3>
              <div
                class="py-2"
                style="font-weight: 400;"
              >
                <p
                  class="text-lg"
                >
                  -- --
                </p>
              </div>
            </div>
          </div>
          <div
            class="my-2"
            id="appendix_k"
          >
            <div
              class="w-full border-[1px] overflow-auto"
            >
              <table
                class="w-full caption-bottom text-sm"
              >
                <thead
                  class="[&_tr]:border-b"
                >
                  <tr
                    class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
                  >
                    <th
                      class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5 w-[300px]"
                    >
                      <div
                        class="flex items-center gap-1"
                      >
                        Package ID
                        <svg
                          class=".1em w-5 h-5 opacity-0"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m5 12 7-7 7 7"
                          />
                          <path
                            d="M12 19V5"
                          />
                        </svg>
                      </div>
                    </th>
                    <th
                      class="px-4 pl-2 text-left font-semibold text-muted-foreground [&:has([role=checkbox])]:pr-0 leading-5"
                    >
                      <div
                        class="flex items-center gap-1"
                      >
                        Actions
                        <svg
                          class=".1em w-5 h-5 opacity-0"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m5 12 7-7 7 7"
                          />
                          <path
                            d="M12 19V5"
                          />
                        </svg>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="[&_tr:last-child]:border-0"
                >
                  <tr
                    class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted text-sm"
                  >
                    <td
                      class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0 font-medium"
                    >
                      <p>
                        withdrawn
                      </p>
                    </td>
                    <td
                      class="px-2 py-3 text-left align-middle text-sm [&:has([role=checkbox])]:pr-0"
                    >
                      <button
                        class="items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-9 rounded-md px-3 flex gap-1"
                      >
                        <svg
                          class="lucide lucide-undo2"
                          fill="none"
                          height="20"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M9 14 4 9l5-5"
                          />
                          <path
                            d="M4 9h10.5a5.5 5.5 0 0 1 5.5 5.5v0a5.5 5.5 0 0 1-5.5 5.5H11"
                          />
                        </svg>
                        Withdraw
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </section>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
